src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -4613,5 +4646,803 @@ class var
+                var compilation = sourceModule.DeclaringCompilation;
+                var tree = compilation.SyntaxTrees.First();
+                var model = compilation.GetSemanticModel(tree);
+
+                var discard2 = GetDiscardDesignations(tree).Skip(1).First();
+                var nestedDeclaration = (DeclarationExpressionSyntax)discard2.Parent.Parent;
+                Assert.Equal("var (_, x3)", nestedDeclaration.ToString());
+                Assert.Null(model.GetDeclaredSymbol(nestedDeclaration));
+                Assert.Null(model.GetDeclaredSymbol(discard2));
+                //Assert.Equal("", model.GetTypeInfo(nestedDeclaration).Type.ToString()); // PROTOTYPE(wildcards) null
+
+                var tuple = (TupleExpressionSyntax)discard2.Parent.Parent.Parent.Parent;
+                Assert.Equal("(var _, var (_, x3))", tuple.ToString());
+                Assert.Equal("(System.String, (System.Int32, System.Int32))", model.GetTypeInfo(tuple).Type.ToTestDisplayString());
+            };
+
+            var comp = CreateCompilationWithMscorlib45(source, parseOptions: TestOptions.Script, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "43", sourceSymbolValidator: validator);
+        }
+
+        [Fact]
+        public void VariousDiscardsInForeach()
+        {
+            var source =
+@"
+class C
+{
+    static void Main()
+    {
+        foreach ((var _, int _, _, var (_, _), int x) in new[] { (1, 2, 3, (4, 5), 6) })
+        {
+            System.Console.Write(x);
+        }
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlib(source, options: TestOptions.DebugExe, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "6");
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var discard1 = GetDiscardDesignations(tree).First();
+            Assert.Null(model.GetDeclaredSymbol(discard1));
+            var declaration1 = (DeclarationExpressionSyntax)discard1.Parent;
+            Assert.Equal("var _", declaration1.ToString());
+            //Assert.Equal("", model.GetTypeInfo(declaration1).Type.ToTestDisplayString()); // PROTOTYPE(wildcards) fix
+
+            var discard2 = GetDiscardDesignations(tree).Skip(1).First();
+            Assert.Null(model.GetDeclaredSymbol(discard2));
+            var declaration2 = (DeclarationExpressionSyntax)discard2.Parent;
+            Assert.Equal("int _", declaration2.ToString());
+            //Assert.Equal("", model.GetTypeInfo(declaration2).Type.ToTestDisplayString()); // PROTOTYPE(wildcards) fix
+
+            var discard3 = GetDiscardIdentifiers(tree).First();
+            Assert.Null(model.GetDeclaredSymbol(discard3));
+        }
+
+        [Fact]
+        public void DiscardInLegacyForeach()
```

Comment:
Refers to regular foreach as opposed to the new component foreach. I'll rename test to `DiscardInCSharp6Foreach`.